File: web/src/api/cloudpods/network/secgroup.js
import request from '@/utils/request'
//获取
export function getSecgroupList(params){
    return request({
        url:'cloud/network/secgroup/getSecgroupList',
        method:'post',
        data: params
    })
}
//创建VPC
export function addSecgroup(params){
    return request({
        url:'cloud/network/secgroup/addSecgroup',
        method:'post',
        data: params
    })
}
//删除VPC
export function deleteSecgroup(params){
    return request({
        url:'cloud/network/secgroup/deleteSecgroup',
        method:'post',
        data: params
    })
}
//批量删除VPC
export function deleteSecgroupByIds(params){
    return request({
        url:'cloud/network/secgroup/deleteSecgroupByIds',
        method:'post',
        data: params
    })
}
//获取
export function getSecgroupRuleList(params){
    return request({
        url:'cloud/network/secgroup/getSecgroupRuleList',
        method:'post',
        data: params
    })
}
//创建VPC
export function addSecgroupRule(params){
    return request({
        url:'cloud/network/secgroup/addSecgroupRule',
        method:'post',
        data: params
    })
}
//创建VPC
export function updateSecgroupRule(params){
    return request({
        url:'cloud/network/secgroup/updateSecgroupRule',
        method:'post',
        data: params
    })
}
//删除VPC
export function deleteSecgroupRule(params){
    return request({
        url:'cloud/network/secgroup/deleteSecgroupRule',
        method:'post',
        data: params
    })
}
//删除VPC
export function deleteSecgroupRuleByIds(params){
    return request({
        url:'cloud/network/secgroup/deleteSecgroupRuleByIds',
        method:'post',
        data: params
    })
}
//创建VPC
export function addSeverSecgroupRule(params){
    return request({
        url:'cloud/network/secgroup/addSeverSecgroupRule',
        method:'post',
        data: params
    })
}
//删除VPC
export function revokeSeverSecgroupRule(params){
    return request({
        url:'cloud/network/secgroup/revokeSeverSecgroupRule',
        method:'post',
        data: params
    })
}

File: server/router/cloudpods/network/secgroup.go
package network

import (
	"github.com/gin-gonic/gin"
	v1 "ygpt/server/api/v1"
)

type NetworkSecgroupRouter struct {
}

func (s *NetworkVpcRouter) InitNetworkSecgroupRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("cloud/network/secgroup")
	secgroupApi := v1.ApiGroupApp.CloudApiGroup.NetApiGroup.NetworkSecgroupApi

	{
		dataRouter.POST("getSecgroupList", secgroupApi.GetSecgroupList)                 // 获取数据列表
		dataRouter.POST("addSecgroup", secgroupApi.AddSecgroup)                         // 新增
		dataRouter.POST("deleteSecgroup", secgroupApi.DeleteSecgroup)                   // 删除
		dataRouter.POST("deleteSecgroupByIds", secgroupApi.DeleteSecgroupByIds)         // 删除
		dataRouter.POST("getSecgroupRuleList", secgroupApi.GetSecgroupRuleList)         // 获取数据列表
		dataRouter.POST("addSecgroupRule", secgroupApi.AddSecgroupRule)                 // 新增
		dataRouter.POST("updateSecgroupRule", secgroupApi.UpdateSecgroupRule)           // 新增
		dataRouter.POST("deleteSecgroupRule", secgroupApi.DeleteSecgroupRule)           // 删除
		dataRouter.POST("deleteSecgroupRuleByIds", secgroupApi.DeleteSecgroupRuleByIds) // 删除

		dataRouter.POST("addSeverSecgroupRule", secgroupApi.AddSeverSecgroupRule)       // 新增
		dataRouter.POST("revokeSeverSecgroupRule", secgroupApi.RevokeSeverSecgroupRule) // 删除
	}
}
